marlowe-runtime/tx/Language/Marlowe/Runtime/Transaction/Submit.hs

```diff
@@ -61,18 +63,38 @@ doSubmit SubmitJobDependencies{..} tellStatus era tx= do
     waitForTx = connectToChainSeek client
       where
         client = ChainSeekClient $ pure clientInit
+
         clientInit = SendMsgRequestHandshake moveSchema ClientStHandshake
           { recvMsgHandshakeRejected = \_ ->
               error "chainseekd schema version mismatch"
-          , recvMsgHandshakeConfirmed = pure clientIdle
+          , recvMsgHandshakeConfirmed = pure clientIdleAwaitConfirmation
           }
-        clientIdle = SendMsgQueryNext (FindTx txId True) clientNext
-        clientNext = ClientStNext
+
+        clientIdleAwaitConfirmation = SendMsgQueryNext (FindTx txId True) clientNextAwaitConfirmation
+
+        clientNextAwaitConfirmation = ClientStNext
           { recvMsgQueryRejected = \err _ ->
               error $ "chainseekd rejected query: " <> show err
-          , recvMsgRollBackward = \_ _ -> pure clientIdle
-          , recvMsgRollForward = \_ point _ -> case point of
+          , recvMsgRollBackward = \_ _ -> pure clientIdleAwaitConfirmation
+          , recvMsgRollForward = \_ point tip -> case (point, tip) of
+              (At block, At tipBlock) -> pure $ clientIdleAwaitMaturity block tipBlock
+              _ -> error "chainseekd rolled forward to genesis"
+          , recvMsgWait = threadDelay 100_000 $> SendMsgPoll clientNextAwaitConfirmation
+          }
+
+        clientIdleAwaitMaturity block tipBlock
+          | blockNo tipBlock - blockNo block >= submitConfirmationBlocks = SendMsgDone block
```

Comment:
Should `WaitForTx` allow the global default confirmation to be overridden by clients?

Reply:
Perhaps, we could add this as an option to the `Submit` command

Reply:
Let's hold off, since we might deprecate `Submit`.